crates/oxc_linter/src/rules/eslint/no_return_await.rs

```diff
@@ -0,0 +1,255 @@
+use oxc_ast::AstKind;
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{Span, GetSpan};
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint(no-return-await): Redundant use of `await` on a return value.")]
+#[diagnostic(severity(warning), help("Remove redundant `await`."))]
+struct NoReturnAwaitDiagnostic(#[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct NoReturnAwait;
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// Disallow unnecessary return await
+    ///
+    /// ### Why is this bad?
+    /// This rule aims to prevent a likely common performance hazard due to a lack of understanding of the semantics of async function.
+    /// https://eslint.org/docs/latest/rules/no-return-await
+    ///
+    /// ### Example
+    /// ```javascript
+    /// async function foo() {
+    ///   return await bar();
+    /// }
+    /// ```
+    NoReturnAwait,
+    correctness
```

Comment:
This is not part of eslint recommended, but the extra `return await` seems useless to me 🤔 Let's keep this as correctness and see if there are false positives.

You can test rule alone with `-A all -D no-return-await` in your internal codebase and see if anything comes up?

Reply:
I will try it later